torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -945,24 +945,42 @@ at::Tensor AtenXlaType::cross(const at::Tensor& self, const at::Tensor& other,
 
 at::Tensor AtenXlaType::cumprod(const at::Tensor& self, int64_t dim,
                                 at::ScalarType dtype) const {
-  return bridge::AtenFromXlaTensor(
-      XLATensor::cumprod(bridge::GetXlaTensor(self), dim, dtype));
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
```

Comment:
In tensor_util.h:
```
inline bool HasNativeSupport(xla::PrimitiveType type, const Device& device) {
  return GetDevicePrimitiveType(type, &device) == type;
}
```

And use that.

Reply:
We don't have an `xla::PrimitiveType` here, we have an `at::ScalarType`. How would this work?

Reply:
```
inline bool HasNativeSupport(at::ScalarType type, const Device& device) {
  return TensorTypeFromXlaType(MakeXlaPrimitiveType(type, &device)) == type;
}
```

Reply:
Obviously! Ok.

Reply:
Done.